In [ ]:
import numpy as np

np.random.seed(42)
height = np.round(np.random.normal(180,10,1000)) # выборка
height

In [44]:
import statistics as stat

stat.mean(height)

np.float64(180.201)

Какова вероятность получить среднее выборки 180.2 см,
При условии истинное среднее генеральной совокупности действительно равно 182 см

In [42]:
import scipy.stats as st

t_statistic, p_valeu = st.ttest_1samp(height,182)
t_statistic
print(p_valeu)

9.035492171563735e-09


In [43]:
if p_valeu < 0.05: # пороговое значение 5% 
    print("Есть основания отвергнуть нулевую гипотезу") 
else: 
    print("Нет оснований отвергнуть нулевую гипотезу")

Есть основания отвергнуть нулевую гипотезу


Доверительный интервал в 90% случаев показывает в каких рамках может быть истинное среднее генеральной совокупности, с таким средним выборки.



In [56]:
ci = st.norm.interval(confidence=0.90, loc=np.mean(height), scale=st.sem(height))

print((round(ci[0], 2), round(ci[1], 2)))

(np.float64(179.69), np.float64(180.71))
